### PR TITLE
ARUHA-961: added validation of cursors availability when resetting cursors

### PR DESCRIPTION
**Pull request by v-stepanov**
[ARUHA-961: Reseting offset to non-existing value leads to confusing behaviour](https://techjira.zalando.net/browse/ARUHA-961)

When cursors are reset we don't check the availability of offsets. We should do that to avoid situations that users bring their subscriptions to corrupted state. And we actually do this validation when creating subscription with initial offsets, so we should be consistent here.

## Review
- [x] Tests
- [ ] Documentation
- [x] CHANGELOG

## Deployment Notes
no

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/service/CursorsService.java [hunk 1]
```diff
@@ -183,6 +178,35 @@ private void validateSubscriptionCursors(final Subscription subscription, final
...
+        final Map<TopicRepository, List<NakadiCursor>> cursorsByRepo = cursors.stream()
+                .collect(Collectors.groupingBy(c -> timelineService.getTopicRepository(c.getTimeline())));
+
+        for (final Map.Entry<TopicRepository, List<NakadiCursor>> repoEntry : cursorsByRepo.entrySet()) {
+            final TopicRepository topicRepository = repoEntry.getKey();
+            final List<NakadiCursor> cursorsForRepo = repoEntry.getValue();
+            try {
+                topicRepository.validateReadCursors(cursorsForRepo);
+            } catch (final InvalidCursorException e) {
+                throw new UnableProcessException(e.getMessage(), e);
```
Comment (lmontrieux) [inline src/main/java/org/zalando/nakadi/service/CursorsService.java]: You don't want to use an exception that extends `MyNakadiRuntimeException1`?

File: src/acceptance-test/java/org/zalando/nakadi/webservice/hila/HilaAT.java [hunk 2]
```diff
@@ -33,26 +15,51 @@
...
+import static org.zalando.nakadi.utils.TestUtils.waitFor;
 import static org.zalando.nakadi.webservice.hila.StreamBatch.MatcherIgnoringToken.equalToBatchIgnoringToken;
 import static org.zalando.nakadi.webservice.hila.StreamBatch.singleEventBatch;
-import org.zalando.nakadi.webservice.utils.NakadiTestUtils;
 import static org.zalando.nakadi.webservice.utils.NakadiTestUtils.commitCursors;
 import static org.zalando.nakadi.webservice.utils.NakadiTestUtils.createEventType;
 import static org.zalando.nakadi.webservice.utils.NakadiTestUtils.createSubscription;
 import static org.zalando.nakadi.webservice.utils.NakadiTestUtils.publishEvent;
 import static org.zalando.nakadi.webservice.utils.NakadiTestUtils.publishEvents;
-import org.zalando.nakadi.webservice.utils.TestStreamingClient;
```
Comment (v-stepanov) [inline src/acceptance-test/java/org/zalando/nakadi/webservice/hila/HilaAT.java]: some old PRs brought wrong import format.
I create a separate PR to address this problem: https://github.com/zalando/nakadi/pull/726

File: src/main/java/org/zalando/nakadi/service/CursorsService.java (hunk 1 above)
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/service/CursorsService.java]: done https://github.com/zalando/nakadi/pull/724/commits/6ae58153c46834ec1246604bec6cb86369cbfe49

**Comment by v-stepanov** [comment]
👍

**Comment by lmontrieux** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Comment by lmontrieux** [comment]
👍

**Pull request merged**